commitchecker/commitchecker.go

```diff
@@ -6,19 +6,78 @@ import (
 	"os"
 )
 
+type FetchMode string
+
+const (
+	HTTPS FetchMode = "https"
+	SSH   FetchMode = "ssh"
+)
+
+type Options struct {
+	Start      string
+	End        string
+	ConfigFile string
+	FetchMode  string
+}
+
+func DefaultOptions() Options {
+	return Options{
+		Start:      "main",
+		End:        "HEAD",
+		ConfigFile: "./commitchecker.yaml",
+		FetchMode:  string(HTTPS),
+	}
+}
+
+func (o *Options) Bind(fs *flag.FlagSet) {
+	fs.StringVar(&o.Start, "start", o.Start, "The start of the revision range for analysis")
+	fs.StringVar(&o.End, "end", o.End, "The end of the revision range for analysis")
+	fs.StringVar(&o.ConfigFile, "config", o.ConfigFile, "The configuration file to use (optional).")
+	fs.StringVar(&o.FetchMode, "fetch-mode", o.FetchMode, "Method to use for fetching from git remotes.")
```

Comment:
This one is actually here because (at least for me, how I have `git` set up on my machine), I want to use `ssh` to fetch, but the way the bots are set up in the CI env they need HTTPS - so it's less a configuration option for the repository writ large (as the config file is) and moreso a runtime choice by the entity running this binary, which is why I added it as a flag.